Python/marshal.c

```diff
@@ -1658,7 +1658,11 @@ r_object(RFILE *p)
         Py_ssize_t idx = r_ref_reserve(flag, p);
         PyObject *stop = NULL;
         PyObject *step = NULL;
-        PyObject *start = r_object(p);
+        PyObject *start = NULL;
+        if (idx < 0) {
```

Comment:
You can move up, right after `r_ref_reserve(flag, p)` call, there's no need to initialize this vars for that.